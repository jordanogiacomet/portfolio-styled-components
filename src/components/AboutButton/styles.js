import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
