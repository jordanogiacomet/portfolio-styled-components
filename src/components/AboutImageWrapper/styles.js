import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
