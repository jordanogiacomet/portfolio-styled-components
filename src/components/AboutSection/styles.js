import styled from 'styled-components';

export const Container = styled.section`
  padding: 50px 0;
  display: flex;
  flex-direction: column;

  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }

  @media (min-width: 1280px) {
    gap: 50px;
  }
`;

