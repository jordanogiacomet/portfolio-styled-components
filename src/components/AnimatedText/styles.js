import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: black;
  }
  &.dark{
    color: white;
  }
`;