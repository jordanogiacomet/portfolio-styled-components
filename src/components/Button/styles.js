import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 