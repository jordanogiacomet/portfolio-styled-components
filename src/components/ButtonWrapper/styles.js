import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;