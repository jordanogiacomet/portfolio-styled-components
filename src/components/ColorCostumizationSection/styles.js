// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
