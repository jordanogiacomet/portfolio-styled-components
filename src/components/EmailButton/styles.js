import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ from, via, to }) => css`
    background: linear-gradient(to bottom right, ${from}, ${via}, ${to});
  `};
  border: transparent;
  color: ${({ theme }) => theme.colors.white};
  font-weight: 600;
  padding: 1rem 1.25rem;
  border-radius: 10rem;
  width: 100%;
  transition: scale 0.2s ease-in-out;
  cursor: pointer;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.6;
  }

  &:hover {
    transition: scale ease-in-out 0.2s;
    scale: calc(1.02);
  }

  &:focus {
    outline: none;
  }
`;
