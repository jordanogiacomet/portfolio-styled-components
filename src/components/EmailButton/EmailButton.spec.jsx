import { screen } from '@testing-library/react';
import { EmailButton } from '.';
import { renderTheme } from '../../styles/render-theme';
import { ColorProvider } from '../../contexts/ColorCustomizationContext';
import { ColorModeProvider } from '../../contexts/ColorModeContext';

describe('<EmailButton />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <EmailButton type="submit">Send</EmailButton>
        </ColorProvider>
      </ColorModeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render EmailButton with correct children', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <EmailButton type="submit">Send</EmailButton>
        </ColorProvider>
      </ColorModeProvider>
    );
    const button = screen.getByText('Send');
    expect(button).toBeInTheDocument();
  });

  it('should have the correct type attribute', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <EmailButton type="submit">Send</EmailButton>
        </ColorProvider>
      </ColorModeProvider>
    );
    const button = screen.getByText('Send');
    expect(button).toHaveAttribute('type', 'submit');
  });
});
