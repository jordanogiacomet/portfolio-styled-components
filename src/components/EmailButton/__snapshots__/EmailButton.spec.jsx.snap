// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailButton /> should match snapshot 1`] = `
.c0 {
  background: linear-gradient(to bottom right, #1E90FF, #4169E1, #00008B);
  border: transparent;
  color: white;
  font-weight: 600;
  padding: 1rem 1.25rem;
  border-radius: 10rem;
  width: 100%;
  transition: scale 0.2s ease-in-out;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c0:hover {
  transition: scale ease-in-out 0.2s;
  scale: calc(1.02);
}

.c0:focus {
  outline: none;
}

<div>
  <button
    class="c0"
    from="#1E90FF"
    to="#00008B"
    type="submit"
    via="#4169E1"
  >
    Send
  </button>
</div>
`;
